ambari-server/src/main/resources/stack-hooks/before-INSTALL/scripts/params.py

```diff
@@ -58,19 +58,19 @@
 #hosts
 hostname = execution_command.get_host_name()
 ambari_server_hostname = execution_command.get_ambari_server_host()
-rm_host = execution_command._execution_command.__get_value("clusterHostInfo/resourcemanager_hosts", [])
-slave_hosts = execution_command._execution_command.__get_value("clusterHostInfo/datanode_hosts", [])
-oozie_servers = execution_command._execution_command.__get_value("clusterHostInfo/oozie_server", [])
-hcat_server_hosts = execution_command._execution_command.__get_value("clusterHostInfo/webhcat_server_hosts", [])
-hive_server_host =  execution_command._execution_command.__get_value("clusterHostInfo/hive_server_hosts", [])
-hbase_master_hosts = execution_command._execution_command.__get_value("clusterHostInfo/hbase_master_hosts", [])
-hs_host = execution_command._execution_command.__get_value("clusterHostInfo/historyserver_hosts", [])
-jtnode_host = execution_command._execution_command.__get_value("clusterHostInfo/jtnode_hosts", [])
-namenode_host = execution_command._execution_command.__get_value("clusterHostInfo/namenode_hosts", [])
-zk_hosts = execution_command._execution_command.__get_value("clusterHostInfo/zookeeper_server_hosts", [])
-ganglia_server_hosts = execution_command._execution_command.__get_value("clusterHostInfo/ganglia_server_hosts", [])
-storm_server_hosts = execution_command._execution_command.__get_value("clusterHostInfo/nimbus_hosts", [])
-falcon_host = execution_command._execution_command.__get_value("clusterHostInfo/falcon_server_hosts", [])
+rm_host = execution_command.get_value("clusterHostInfo/resourcemanager_hosts", [])
+slave_hosts = execution_command.get_value("clusterHostInfo/datanode_hosts", [])
+oozie_servers = execution_command.get_value("clusterHostInfo/oozie_server", [])
+hcat_server_hosts = execution_command.get_value("clusterHostInfo/webhcat_server_hosts", [])
+hive_server_host =  execution_command.get_value("clusterHostInfo/hive_server_hosts", [])
+hbase_master_hosts = execution_command.get_value("clusterHostInfo/hbase_master_hosts", [])
+hs_host = execution_command.get_value("clusterHostInfo/historyserver_hosts", [])
+jtnode_host = execution_command.get_value("clusterHostInfo/jtnode_hosts", [])
+namenode_host = execution_command.get_value("clusterHostInfo/namenode_hosts", [])
+zk_hosts = execution_command.get_value("clusterHostInfo/zookeeper_server_hosts", [])
+ganglia_server_hosts = execution_command.get_value("clusterHostInfo/ganglia_server_hosts", [])
+storm_server_hosts = execution_command.get_value("clusterHostInfo/nimbus_hosts", [])
+falcon_host = execution_command.get_value("clusterHostInfo/falcon_server_hosts", [])
 
 has_sqoop_client = 'sqoop-env' in module_configs
 has_namenode = not len(namenode_host) == 0
```

Comment:
This line will also fail since the field is late-accessed by config_dictionary
```
Traceback (most recent call last):
  File "/var/lib/ambari-agent/cache/stack-hooks/before-INSTALL/scripts/hook.py", line 37, in <module>
    BeforeInstallHook().execute()
  File "/usr/lib/ambari-agent/lib/resource_management/libraries/script/script.py", line 349, in execute
    method(env)
  File "/var/lib/ambari-agent/cache/stack-hooks/before-INSTALL/scripts/hook.py", line 28, in hook
    import params
  File "/var/lib/ambari-agent/cache/stack-hooks/before-INSTALL/scripts/params.py", line 78, in <module>
    has_namenode = not len(namenode_host) == 0
  File "/usr/lib/ambari-agent/lib/resource_management/libraries/script/config_dictionary.py", line 73, in __getattr__
    raise Fail("Configuration parameter '" + self.name + "' was not found in configurations dictionary!")
```
```